### PR TITLE
OCM-853 | feat: day1/2 operations for managed ingress attributes

### PR DESCRIPTION
**Pull request by gdbranco**
Related issue: https://issues.redhat.com/browse/OCM-853

Day 2
```
rosa edit ingress -c 24qu5qv2ips5j2bhsan1kv0osagkq6pb apps
? Label match for ingress (optional): foo=bar
? Excluded namespaces for ingress (optional): stage,dev,int
? Private ingress (optional): No
? Wildcard Policy (optional, choose 'none' to skip selection): WildcardsAllowed
? Namespace Ownership Policy (optional, choose 'none' to skip selection): InterNamespaceAllowed
? Cluster Routes Hostname (optional): 
? Cluster Routes TLS Secret Reference (optional): 
I: Updated ingress 'f8g2' on cluster '24qu5qv2ips5j2bhsan1kv0osagkq6pb'
```

Syncset
```
 kind: IngressController
      name: default
      namespace: openshift-ingress-operator
      patch: >-
        {"spec":{"defaultCertificate":{"name":"gdb-413-primary-cert-bundle-secret"},"domain":"apps.gdb-413.caas.i1.devshift.org","endpointPublishingStrategy":{"loadBalancer":{"providerParameters":{"aws":{"type":"NLB"},"type":"AWS"},"scope":"External"},"type":"LoadBalancerService"},"namespaceSelector":{"matchExpressions":[{"key":"kubernetes.io/metadata.name","operator":"NotIn","values":["stage","dev","int"]}]},"routeAdmission":{"namespaceOwnership":"InterNamespaceAllowed","wildcardPolicy":"WildcardsAllowed"},"routeSelector":{"matchExpressions":[{"key":"foo","operator":"NotIn","values":["bar"]}]}}}
```

Day 1
```
rosa create cluster --sts \
--cluster-name gdb-413 \
--role-arn arn:aws:iam::xxx:role/ManagedOpenShift-Installer-Role \
--controlplane-iam-role arn:aws:iam::xxx:role/ManagedOpenShift-ControlPlane-Role \
--support-role-arn arn:aws:iam::xxx:role/ManagedOpenShift-Support-Role \
--worker-iam-role arn:aws:iam::xxx:role/ManagedOpenShift-Worker-Role --mode auto -y --version 4.13.1 --fake-cluster --default-ingress-route-selector="foo=bar" --default-ingress-excluded-namespaces="stage,int,dev" --default-ingress-wildcard-policy="WildcardsAllowed"
```

Syncset
```
patch: >-
        {"spec":{"defaultCertificate":{"name":"gdb-413-primary-cert-bundle-secret"},"domain":"apps.gdb-413.9e04.i1.devshift.org","endpointPublishingStrategy":{"loadBalancer":{"providerParameters":{"aws":{"type":"NLB"},"type":"AWS"},"scope":"External"},"type":"LoadBalancerService"},"namespaceSelector":{"matchExpressions":[{"key":"kubernetes.io/metadata.name","operator":"NotIn","values":["stage","int","dev"]}]},"routeAdmission":{"namespaceOwnership":"Strict","wildcardPolicy":"WildcardsAllowed"},"routeSelector":{"matchExpressions":[{"key":"foo","operator":"NotIn","values":["bar"]}]}}}
```

### CODE REVIEW

File: cmd/edit/ingress/cmd.go [hunk 1]
```diff
@@ -177,24 +273,114 @@ func run(cmd *cobra.Command, argv []string) {
...
+		}
+		clusterRoutesTlsSecretRef = &args.clusterRoutesTlsSecretRef
+	} else if interactive.Enabled() && !ocm.IsHyperShiftCluster(cluster) {
+		clusterRoutesTlsSecretRefArg, err := interactive.GetString(interactive.Input{
+			Question: "Cluster Routes Hostname",
+			Help:     cmd.Flags().Lookup(clusterRoutesTlsSecretRefFlag).Usage,
+			Default:  args.clusterRoutesTlsSecretRef,
+		})
+		if err != nil {
+			r.Reporter.Errorf("Expected a valid Cluster Route Hostname: %s", err)
```
Comment (thomasmckay) [inline cmd/edit/ingress/cmd.go:380]: should be "Cluster Routes Hostname" (plural)

File: cmd/edit/ingress/cmd.go [hunk 2]
```diff
@@ -134,23 +208,25 @@ func run(cmd *cobra.Command, argv []string) {
...
 			os.Exit(1)
 		}
 		labelMatch = &args.labelMatch
 	} else if interactive.Enabled() && !ocm.IsHyperShiftCluster(cluster) {
 		labelMatchArg, err := interactive.GetString(interactive.Input{
-			Question: "Label match for ingress",
+			Question: "Route Selector for ingress",
 			Help:     cmd.Flags().Lookup(labelMatchFlag).Usage,
-			Default:  labelMatch,
+			Default:  args.labelMatch,
```
Comment (hunterkepley) [inline cmd/edit/ingress/cmd.go:229]: NP: Little confusing to me, having the default still be `args.labelMatch` and this being Route Selector. WDYT?

**Review by hunterkepley — COMMENTED** [review]
Overall looks good, just one nitpick

File: cmd/create/cluster/cmd.go [hunk 3]
```diff
@@ -2923,6 +3066,27 @@ func buildCommand(spec ocm.Spec, operatorRolesPrefix string,
 		command += fmt.Sprintf(" --audit-log-arn %s", *spec.AuditLogRoleARN)
 	}
 
+	if !reflect.DeepEqual(spec.DefaultIngress, ocm.DefaultIngressSpec{}) {
+		if len(spec.DefaultIngress.RouteSelectors) != 0 {
+			command += fmt.Sprintf(" --%s \"", defaultIngressRouteSelectorFlag)
+			for k, v := range spec.DefaultIngress.RouteSelectors {
+				command += fmt.Sprintf("%s=%s", k, v)
+			}
+			command += "\""
```
Comment (renan-campos) [inline cmd/create/cluster/cmd.go:3075]: If there is more than one key value pair, a comma separated list is not properly made here.

Comment (renan-campos) [reply cmd/create/cluster/cmd.go:3075]: Example (account id omitted)
```
? Route Selector for ingress (optional): foo=bar,test=case                                                                                         
? Excluded namespaces for ingress (optional): alpha,beta,gamma,delta                                                                               
? Wildcard Policy (optional, choose 'none' to skip selection): WildcardsDisallowed                                                                 
? Namespace Ownership Policy (optional, choose 'none' to skip selection): Strict                                                                   
I: Creating cluster 'rc-test'                                                                                                                      
I: To create this cluster again in the future, you can run:                                                                                        
   rosa create cluster --cluster-name rc-test --sts --role-arn arn:aws:iam::~:role/ManagedOpenShift-Installer-Role --support-role-arn ar
n:aws:iam::~:role/ManagedOpenShift-Support-Role --controlplane-iam-role arn:aws:iam::~:role/ManagedOpenShift-ControlPlane-Rol
e --worker-iam-role arn:aws:iam::~/ManagedOpenShift-Worker-Role --operator-roles-prefix rc-test-k9a2 --region us-east-1 --version 4
.12.22 --replicas 2 --compute-machine-type m5.xlarge --machine-cidr 10.0.0.0/16 --service-cidr 172.30.0.0/16 --pod-cidr 10.128.0.0/14 --host-prefix
 23 --default-ingress-route-selector "foo=bartest=case" --default-ingress-excluded-namespaces alpha,beta,gamma,delta --default-ingress-wildcard-pol
icy WildcardsDisallowed --default-ingress-namespace-ownership-policy Strict
```

File: cmd/create/cluster/cmd.go [hunk 4]
```diff
@@ -2923,6 +3066,27 @@ func buildCommand(spec ocm.Spec, operatorRolesPrefix string,
...
+			command += fmt.Sprintf(" --%s %s", defaultIngressExcludedNamespacesFlag,
+				strings.Join(spec.DefaultIngress.ExcludedNamespaces, ","))
+		}
+		if spec.DefaultIngress.WildcardPolicy != "" {
+			command += fmt.Sprintf(" --%s %s", defaultIngressWildcardPolicyFlag, spec.DefaultIngress.WildcardPolicy)
+		}
+		if spec.DefaultIngress.NamespaceOwnershipPolicy != "" {
+			command += fmt.Sprintf(" --%s %s", defaultIngressNamespaceOwnershipPolicyFlag,
+				spec.DefaultIngress.NamespaceOwnershipPolicy)
+		}
```
Comment (renan-campos) [inline cmd/create/cluster/cmd.go:3087]: If none was selected, the built command has this flag without an argument, which causes the command to fail. 
```
Failed to execute root command: flag needs an argument: --default-ingress-namespace-ownership-policy
```
This is likewise the case with the other option that has a 'none' option as well.

**Review by renan-campos — COMMENTED** [review]
Found some minor bugs in the create cluster code with the build command. Haven't tried the edit ingress command yet. Overall code looks good.

File: cmd/create/cluster/cmd.go (hunk 3 above)
Comment (gdbranco) [reply cmd/create/cluster/cmd.go:3075]: Oh, missed the comma lol. Thanks!

File: cmd/create/cluster/cmd.go (hunk 4 above)
Comment (gdbranco) [reply cmd/create/cluster/cmd.go:3087]: Thanks for catching it, I added the printing commands at a later moment

**Review by renan-campos — APPROVED** [review]
/lgtm

**Pull request merged**